### PR TITLE
Updating fluentd.conf to properly set date for syslog lines and 

### PR DESCRIPTION
**Pull request by ewolinetz**
use corrected time for index name

### CODE REVIEW

**Comment by ewolinetz** [comment]
Addresses #46 

@sosiouxme @jcantrill @richm PTAL

**Comment by ewolinetz** [comment]
I think there is an issue that I didn't encounter yesterday but am today; please do not merge yet

**Comment by ewolinetz** [comment]
Needed to update gsub function to use 
```
(tag_parts[3].nil? ? Time.at(time).year.to_s : tag_parts[3][9,4]) )
```
instead of `||` logic

**Comment by portante** [comment]
@ewolinetz, so this file description talks about reading logs from docker logs, but I see two sources, one for /var/log/messages* and one for container logs.  Seems like we should address that with these comment changes, since we appear to be rewriting large sections of the comments.

It also seems like we are doing a lot of work in fluentd to handle rsyslog data when I believe rsyslog can do that much easier, like dealing with proper timestamps and index names, etc.  Then you have a fluentd for processing container logs, and an rsyslog config for system logs (which it is already handling, not extra data flow).

**Comment by richm** [comment]
@portante I am trying to figure out how we can use rsyslog instead of fluentd for this use case, so that we do not have to have fluentd or fluentd+rsyslog running on each node.  We can just have rsyslog, and it can either send the logs directly to elasticsearch, to a message queue, or to some other forwarder/aggregator.

**Comment by ewolinetz** [comment]
@portante the trello card regarding using rsyslog instead is this one: https://trello.com/c/oUvJg3JR/188-r-d-use-rsyslog-instead-of-fluentd-logging

**Comment by portante** [comment]
I would suggest that the we just make a simple change (when we are ready) to the `rsyslog.conf` to get `/var/log/messages` to use a file format with high resolution time stamps (and timezone information): `RSYSLOG_FileFormat`, see http://www.rsyslog.com/doc/v7-stable/configuration/templates.html (near the bottom).

This is a simple change.

File: fluentd/fluent.conf [hunk 1]
```diff
@@ -1,111 +1,111 @@
...
+# installation of the fluent-plugin-elasticsearch & the
+# fluent-plugin-kubernetes_metadata_filter plugins.
+# See https://github.com/uken/fluent-plugin-elasticsearch &
+# https://github.com/fabric8io/fluent-plugin-kubernetes_metadata_filter for
+# more information about the plugins.
+# Maintainer: Jimmi Dyson <jimmidyson@gmail.com>
+#
+# Example
+# =======
+# A line in the Docker log file might like like this JSON:
```
Comment (sosiouxme) [inline fluentd/fluent.conf]: "look like"

File: fluentd/fluent.conf [hunk 2]
```diff
@@ -1,111 +1,111 @@
...
+# Kubernetes pod has been restarted (resulting in a several Docker container IDs).
+#
+# TODO: Propagate the labels associated with a container along with its logs
+# so users can query logs using labels as well as or instead of the pod name
+# and container name. This is simply done via configuration of the Kubernetes
 # fluentd plugin but requires secrets to be enabled in the fluent pod. This is a
-# problem yet to be solved as secrets are not usable in static pods which the fluentd         
-# pod must be until a per-node controller is available in Kubernetes.                                                                                                     
+# problem yet to be solved as secrets are not usable in static pods which the fluentd
+# pod must be until a per-node controller is available in Kubernetes.
```
Comment (sosiouxme) [inline fluentd/fluent.conf]: Above paragraph no longer relevant?

File: fluentd/fluent.conf [hunk 3]
```diff
@@ -228,7 +228,9 @@
       # if host == 'localhost' do a fqdn lookup
       ## we pull the hostname from the host's /etc/hostname mounted at /etc/docker-hostname to correctly identify the hostname
       hostname ${host.eql?('localhost') ? (begin; File.open('/etc/docker-hostname') { |f| f.readline }.rstrip; rescue; host; end) : host}
-      time ${Time.at(time).to_datetime.to_s}
+
+      # tag_parts = system.var.log.messages-20150710
+      time ${ Time.at(time) > Time.now ? (temp_time = Time.parse(Time.at(time).to_s.gsub(Time.at(time).year.to_s, (tag_parts[3].nil? ? Time.at(time).year.to_s : tag_parts[3][9,4]) )).to_datetime.to_s; Time.parse(temp_time) > Time.now ? Time.parse(temp_time.gsub(Time.parse(temp_time).year.to_s, (Time.parse(temp_time).year - 1).to_s )).to_datetime.to_s : Time.parse(temp_time).to_datetime.to_s ) : Time.at(time).to_datetime.to_s }
```
Comment (sosiouxme) [inline fluentd/fluent.conf]: Clarify this if possible; could also use a comment as to why this is needed

File: fluentd/fluent.conf (hunk 2 above)
Comment (ewolinetz) [inline fluentd/fluent.conf]: I'll have to check again on whether or not we are sending labels, I feel like we are but I'd like to be sure before we scrap that paragraph entirely

**Comment by ewolinetz** [comment]
@sosiouxme addressed comments

**Comment by richm** [comment]
lgtm

**Pull request merged**